shared/SideMenu.js

```diff
@@ -0,0 +1,29 @@
+import "jsdom";
+export default {
+	injectHeaderID (content) {
+		const JSDOM = require("jsdom").JSDOM;
+
+		const jsdom = new JSDOM(content);
+		const { window } = jsdom;
+
+		const h2List = window.document.getElementsByTagName("h2");
+		for (let i = 0; i < h2List.length; i++) {
+			h2List[i].setAttribute("id", `${h2List[i].textContent.replace(/\s+/g, "-")}`);
+		}
+		return jsdom.serialize();
+	},
+	getHeaderList (content) {
+		const JSDOM = require("jsdom").JSDOM;
+		const jsdom = new JSDOM(content);
+		const { window } = jsdom;
+
+		let headerList = [...window.document.getElementsByTagName("h2")];
```

Comment:
`window.document.getElementsByTagName("h2")` should already return an array. Is it necessary to reprocess it with `[...h2Array]`?

Reply:
Yes, it is necessary. `window.document.getElementsByTagName("h2")` returns a collection. You cannot do array methods like `.map` on collections only Arrays.

Reply:
Couldn't you just use [Array.from](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)?


```
let h2Collection = window.document.getElementsByTagName("h2");
return Array.from(h2Collection, function (header) {
     return {
          title: header.textContent,
          href: `#${header.textContent.replace(/\s+/g, "-")}`
     };
});
```

Reply:
Yes, I believe I could. should I change the code to use it?